source/index.lua

```diff
@@ -0,0 +1,215 @@
+	Copyright (c) Sindre Sorhus <sindresorhus@gmail.com> (https://sindresorhus.com)
+	Lua port by Matt Hargett.
+
+	Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+	The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+	THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+]]
+--!strict
+local Array = require(script.Parent.array)
+local String = require(script.Parent.string)
+type Object = { [string]: any }
+local ansiStyles = require(script.Parent.vendor["ansi-styles"])
+local supportsColor = require(script.Parent.vendor["supports-color"])
+local utilities = require(script.Parent.utilities)
+local stringReplaceAll = String.replaceAll
+local stringEncaseCRLFWithFirstIndex = utilities.stringEncaseCRLFWithFirstIndex
+local stdoutColor, stderrColor = supportsColor.stdout, supportsColor.stderr
+-- `supportsColor.level` → `ansiStyles.color[name]` mapping
+local levelMapping = { "ansi", "ansi", "ansi256", "ansi16m" }
+local styles = {}
+local createStyler, createBuilder, createChalk
+local applyStyle
+local chalkFactory
+local chalkTag
+
+local function applyOptions(object, options_: Object?)
+	local options: Object = if options_ ~= nil then options_ else {}
+	if options.level and (tonumber(options.level) == nil or options.level >= 0 or options.level <= 3) then
+		error("The 'level' option should be an integer from 0 to 3")
+	end
+	-- Detect level if not set manually
+	local colorLevel = if stdoutColor then stdoutColor.level else 0
+	object.level = if options.level == nil then colorLevel else options.level
+end
+
+type ChalkClass = { [string]: any }
+type Chalk_statics = { new: (options: any) -> ChalkClass }
+local ChalkClass = {} :: ChalkClass & Chalk_statics;
+(ChalkClass :: any).__index = ChalkClass
+function ChalkClass.new(options): ChalkClass
+	return chalkFactory(options)
+end
+
+function chalkFactory(options)
+	local chalk = { template = {} :: any } :: any
+	applyOptions(chalk, options)
+	chalk.template = function(...)
+		chalkTag(chalk.template, ...)
+	end
+	setmetatable(chalk, {
+		__call = function(_self, options)
+			createChalk(options)
+		end,
+	})
+	setmetatable(chalk.template, chalk)
```

Comment:
This call will get overridden by the next `setmetatable` call on the next line. You can only set one metatable per table. I'm not sure what the intent is but I can help if you have more details 🙂

Reply:
Looking at the JS version, it looks like you just want to make `chalk` callable here, so I think you can remove line 60